mex/backend/auxiliary/ldap.py

```diff
@@ -0,0 +1,82 @@
+from functools import cache
+from typing import Annotated
+
+from fastapi import APIRouter, Query
+
+from mex.backend.auxiliary.models import AuxiliarySearch
+from mex.common.ldap.extract import (
+    get_count_of_found_persons_by_name,
+    get_persons_by_name,
+)
+
+# from mex.backend.serialization import to_primitive  # noqa: ERA001
+from mex.common.models import ExtractedPerson, ExtractedPrimarySource
+from mex.common.primary_source.extract import extract_seed_primary_sources
+from mex.common.primary_source.transform import (
+    get_primary_sources_by_name,
+    transform_seed_primary_sources_to_extracted_primary_sources,
+)
+
+router = APIRouter()
+from mex.common.ldap.transform import (  # noqa: E402
+    transform_ldap_persons_to_mex_persons,
+)
+from mex.common.models import ExtractedOrganizationalUnit  # noqa: E402
+from mex.common.organigram.extract import extract_organigram_units  # noqa: E402
+from mex.common.organigram.transform import (  # noqa: E402
+    transform_organigram_units_to_organizational_units,
+)
+
+
+@router.get("/ldap", tags=["editor"])
+def search_person_in_ldap(
+    q: Annotated[str, Query(min_length=1, max_length=1000)],
+    offset: Annotated[int, Query(ge=0, le=10e10)] = 0,
+    limit: Annotated[int, Query(ge=1, le=100)] = 10,
+) -> AuxiliarySearch[ExtractedPerson]:
+    """Search for persons in LDAP by string.
+
+    Args:
+        q: The name of the person to be searched.
+        offset: The starting index for pagination
+        limit: The maximum number of results to return
+
+    Returns:
+        paginated list of ExtractedPerson and the total count of persons found.
+    """
+    count_persons = get_count_of_found_persons_by_name(q)
+    found_persons = list(get_persons_by_name(q))
```

Comment:
this will only search by `surname`

Reply:
actually no. in` ldap/extract.py` I allow to apply the search string on each field: `def get_persons_by_name(
    surname: str = "*",
    given_name: str = "*",
    **filters: str`

Reply:
but how could that work?

the signature of `get_persons_by_name` is:

```python
def get_persons_by_name(
    surname: str = "*",
    given_name: str = "*",
    **filters: str,
) -> Generator[LDAPPerson, None, None]:
    ...
```

if you call it like `get_persons_by_name(q)`, you will only fill the `surname` argument, but never the `given_name`

Reply:
Ah I see what you mean. Hmm, how would like to handle it? Adding an additional param for the givenname or handling the query syntax like `q="family-name:{famName]+AND+given-name:{givName}"` ?

Reply:
there is also a property `displayName` on ldap persons, which will be the full name of a person including all family and given names. it should be safe to search through those. luckily the get_persons_by_name also accepts arbitrary filter arguments, so you can use that directly here:

```python
found_persons = list(get_persons_by_name(displayName=q))
```